csrc/host_ir/jit.cpp

```diff
@@ -68,185 +77,281 @@ T throwIfError(llvm::Expected<T>&& E) {
+    llvm::orc::SymbolMap& symbolMap,
+    llvm::orc::MangleAndInterner& mangler,
+    std::string_view func_name) {
+  auto addr = llvm::orc::ExecutorAddr::fromPtr(func_ptr);
+  symbolMap[mangler(func_name)] =
+      llvm::orc::ExecutorSymbolDef(addr, llvm::JITSymbolFlags::Exported);
+}
+
+// NOTE: this is just a simple example of allocate a output tensor and set it to
+// input tensor The whole concept to to demonstrate llvm jit works, we will
+// change this in the future
+void compileLoadStoreOp(
+    LoadStoreOp* load_store_op,
+    llvm::IRBuilder<>& builder,
+    std::unordered_map<Val*, llvm::Value*>& val_to_value) {
   NVF_ERROR(
-      !llvm::verifyModule(*mod, &error_stream),
-      "LLVM module verification failed: " + error);
+      load_store_op->opType() == LoadStoreOpType::Set ||
+      load_store_op->opType() == LoadStoreOpType::SegmenterSet);
+  NVF_ERROR(
+      load_store_op->out()->isA<TensorView>(), "out must be a TensorView");
+  auto* in_tv = load_store_op->in()->as<Val>();
+  auto* out_tv = load_store_op->out()->as<Val>();
+  auto it = val_to_value.find(in_tv);
+  NVF_ERROR(
+      it != val_to_value.end(), "input tensor is not found in val_to_value");
+  llvm::Module* mod = builder.GetInsertBlock()->getParent()->getParent();
+  llvm::Value* in_tensor_ptr = it->second;
+
+  // allocate a new tensor
+  llvm::Function* allocate_tensor_func =
+      mod->getFunction(kAllocateTensorFuncName);
+  llvm::Value* out_tensor_ptr =
+      builder.CreateCall(allocate_tensor_func, {}, "out_tensor_raw");
+
+  // set the output tensor to the input tensor
+  llvm::Function* set_tensor_func = mod->getFunction(kSetTensorFuncName);
+  builder.CreateCall(set_tensor_func, {out_tensor_ptr, in_tensor_ptr});
+
+  // bind the output tensor to val_to_value
+  val_to_value[out_tv] = out_tensor_ptr;
+
+  if (isDebugDumpEnabled(DebugDumpOption::HostIrJit)) {
+    auto* func = builder.GetInsertBlock()->getParent();
+    llvm::outs() << "=== LLVM IR Generation for LoadStoreOp ===\n";
+    func->print(llvm::outs(), nullptr);
+  }
+}
 
-  const bool debug_print = isDebugDumpEnabled(DebugDumpOption::HostIrJit);
-  if (debug_print) {
-    llvm::outs() << "=== LLVM IR ===\n";
-    mod->print(llvm::outs(), nullptr);
+void compileMainFuncInputs(
+    const hir::HostIrContainer* container,
+    llvm::IRBuilder<>& builder,
+    std::unordered_map<Val*, llvm::Value*>& val_to_value) {
+  llvm::LLVMContext& ctx = builder.getContext();
+
+  // Get the current function (main) and its first argument
+  llvm::Function* func = builder.GetInsertBlock()->getParent();
+  llvm::Value* aten_tensor_array_ptr = func->getArg(0);
+
+  llvm::Type* aten_tensor_array_type = getInt8PtrDynamicArrayType(ctx);
+  // bind input aten tensor sizes to val_to_value
+  for (size_t i = 0; i < container->inputs().size(); ++i) {
+    auto* input = container->inputs()[i];
+    if (TensorView* tv = dynamic_cast<TensorView*>(input)) {
+      llvm::Value* aten_tensor_ptr_addr = builder.CreateGEP(
```

Comment:
```suggestion
      llvm::Value* tensor_addr = builder.CreateGEP(
```

Reply:
The LLVM IR we generate never uses `Tensor` the value, so the `_ptr` suffix is unnecessary. 

Also, the LLVM IR we generate never uses tensors other than at::Tensor, so the `aten_` prefix is unnecessary.